File: src/main/Java/boj_step14/boj10815.java
import java.io.*;
import java.util.Arrays;
import java.util.HashSet;
import java.util.LinkedHashSet;
import java.util.stream.Collectors;

public class boj10815 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        reader.readLine();
        HashSet<Integer> cardSet = Arrays.stream(reader.readLine().split(" ")).map(Integer::parseInt).collect(Collectors.toCollection(HashSet::new));
        int m = Integer.parseInt(reader.readLine());
        LinkedHashSet<Integer> cardKeySet = Arrays.stream(reader.readLine().split(" ")).map(Integer::parseInt).collect(Collectors.toCollection(LinkedHashSet::new));
        for (int card: cardKeySet) writer.append((cardSet.contains(card)) ? "1 " : "0 ");
        writer.close();
    }
}

File: src/main/Kotlin/boj_step14/boj10815.kt
import java.awt.event.KeyAdapter
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    val cards = r.readLine().split(" ").map { it.toInt() }.toMutableSet()
    val m = r.readLine().toInt()
    val keys = r.readLine().split(" ").map { it.toInt() }
    keys.forEach {
        if (!cards.add(it)) w.write("1 ") else w.write("0 ")
    }
    w.close()
}